Makefile

```diff
@@ -9,6 +9,12 @@ test: install | emily-client
 	cargo test \
 		&& pnpm --recursive test
 
+integration-test: install | emily-client
+	docker compose --file docker-compose.test.yml up --detach
+	cargo test --all-features \
+		&& pnpm --recursive test
+	docker compose --file docker-compose.test.yml down
```

Comment:
> Personally, I'd prefer to have the environment set up be more explicit and not hidden behind a `make integration-test` call. Alternatively have a more verbose name for this: `make integration-test-with-docker-environment` (not the best suggestion but you get the idea).

Yeah, maybe we should all chat about this on Friday?


> Also, should we figure out a way to run these integration tests in the CI? That can definitely be done as a follow-up, since I expect the environment setup should be done differently and not with docker-compose within the actual test runner.

We "could" using [GitHub action services](https://docs.github.com/en/actions/using-containerized-services/about-service-containers), but last I checked they do not expose enough options for us to mount a file on the container like we do with docker for `bitcoin.conf`. Plus I think it's better to have CI do the exact same thing that we do locally.

Reply:
Okay interesting. I'm okay with this for now as well. I can see that it is convenient, and for me it's easy to run the integration tests explicitly as well if I prefer.